File: frontend/types/invitation.ts
import { UserRole } from "./user";

export interface Invitation {
  id: string;
  email: string;
  role: UserRole;
  expires_at: string;
  accepted_at: string | null;
  canceled_at: string | null;
  created_at: string;
  created_by: string;
  devLink?: string;
}

export interface InvitationCreate {
  email: string;
  role: UserRole;
}

export interface InvitationAccept {
  token: string;
  password: string;
  full_name: string;
  region?: string;
}

export interface InvitationVerifyResponse {
  valid: boolean;
  email: string;
  role: string;
}

File: backend/app/schemas/invitation.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr

from app.models.user import UserRole


class InvitationCreate(BaseModel):
    email: EmailStr
    role: UserRole


class InvitationAccept(BaseModel):
    token: str
    password: str
    full_name: str
    region: Optional[str] = None


class InvitationResponse(BaseModel):
    id: uuid.UUID
    email: str
    role: UserRole
    expires_at: datetime
    accepted_at: Optional[datetime]
    canceled_at: Optional[datetime]
    created_at: datetime
    created_by: uuid.UUID

    model_config = {"from_attributes": True}


class InvitationVerifyResponse(BaseModel):
    valid: bool
    email: Optional[str] = None
    role: Optional[UserRole] = None
